systemverilog-plugin/UhdmAst.cc

```diff
@@ -4023,6 +4023,9 @@ void UhdmAst::process_tagged_pattern()
         if (field_name != "default") { // TODO: better support of the default keyword
             auto field = new AST::AstNode(static_cast<AST::AstNodeType>(AST::Extended::AST_DOT));
             field->str = field_name;
+            if (find_ancestor({AST::AST_PARAMETER, AST::AST_LOCALPARAM})) {
+                current_node->children[0]->str = current_node->children[0]->str + "." + field_name;
```

Comment:
I think that instead of adding ``.`` here, we should adapt ``UhdmAst::process_assignment_pattern_op``. It looks like ``UhdmAst::process_assignment_pattern_op`` never got updated to new flow with ``AST_DOT`` and is still looking for ``.`` in string. Instead it should check if node have ``AST_DOT`` as children and take ``key`` as string from it.

https://github.com/chipsalliance/yosys-f4pga-plugins/blob/3a440c4d33e47849e4d974f0bd65bb40272716e9/systemverilog-plugin/UhdmAst.cc#L3590-L3591

Example AST that gets ``UhdmAst::process_assignment_pattern_op``:
```
AST_ASSIGN <dut.sv:0.0-0.0> [0x55dd211d51b0]               
  AST_IDENTIFIER <dut.sv:0.0-0.0> [0x55dd21194140] str='\my_package::my_parameter' 
    AST_DOT <dut.sv:0.0-0.0> [0x55dd21221bb0] str='a'      
  AST_CONSTANT <dut.sv:0.0-0.0> [0x55dd21195210] bits='1'(1) range=[0:0] int=1
```

Reply:
Thanks! I applied this modification in the newest commit.